File: src/designpatterns/builder/BuilderJava/HealthyBuilder.java
package designpatterns.builder.BuilderJava;

public class HealthyBuilder extends Builder {

    @Override
    public Builder addStarter() {
        meal.setStart(Starter.SOUP);
        return this;
    }

    @Override
    public Builder addMain() {
        meal.setMain(Main.FISH);
        return this;
    }

    @Override
    public Builder addDessert() {
        meal.setDessert(Dessert.FRUIT_SALAD);
        return this;
    }

    @Override
    public Builder addDrink() {
        meal.setDrink(Drink.FRUIT_JUICE);
        return this;
    }
}

File: src/designpatterns/builder/healthy_builder.cpp
#include "healthy_builder.hpp"
#include "meal.hpp"

Builder* HealthyBuilder::addStarter() {
    meal.setStart(SOUP);
    return this;
}

Builder* HealthyBuilder::addMain() {
    meal.setMain(FISH);
    return this;
}

Builder* HealthyBuilder::addDessert() {
    meal.setDessert(FRUIT_SALAD);
    return this;
}

Builder* HealthyBuilder::addDrink() {
    meal.setDrink(FRUIT_JUICE);
    return this;
}